ihatemoney/web.py

```diff
@@ -640,7 +642,60 @@ def change_lang(lang):
 def settle_bill():
     """Compute the sum each one have to pay to each other and display it"""
     bills = g.project.get_transactions_to_settle_bill()
-    return render_template("settle_bills.html", bills=bills, current_view="settle_bill")
+    bill_forms = [get_payment_form(bill) for bill in bills]
+
+    return render_template(
+        "settle_bills.html",
+        bills=bills,
+        bill_forms=bill_forms,
+        current_view="settle_bill"
+    )
+
+
+def get_payment_form(settlement_bill, **kwargs):
+    form = get_billform_for(g.project, set_default=False, bill_type=BillTypes.settlement.value)
+    bill = Bill()
+    bill.payer_id = settlement_bill['ower'].id
+    bill.amount = settlement_bill['amount']
+    bill.what = _("Refund")
+    bill.date = datetime.now()
+    bill.owers = [settlement_bill['receiver']]
+
+    form.fill(bill)
+    receiver_name = [user
+                     for (id, user)
+                     in form.payed_for.choices
+                     if id == form.payed_for.data[0]
+                     ][0]
```

Comment:
This looks quite ugly to me. Isn't there a better way to write this? (in a more readable way)

Reply:
Is it better like this ?
```
receiver_id = settlement_bill['receiver']
users = form.payed_for.choices
receiver_name = [name for (id, name) in users if id == receiver_id][0]
```

Reply:
If you want to access the users by IDs, then one simple way would be to use a dict (easier to read) :
```python
receiver_id = settlement_bill['receiver']
users = dict(form.payed_for.choices)
receiver_name = users[receiver_id]
```